peregrine/resources/submission/graphql/node.py

```diff
@@ -217,13 +217,27 @@ def apply_arg_quicksearch(q, args, info):
 
 
 def apply_query_args(q, args, info):
+    """
+    Args:
+        q: psqlgraph query
+        args: dictionary of the arguments passed to the query.
+        info: graphene object that holds the query's arguments, models and requested fields.
+    """
+
     pg_props = set(getattr(q.entity(), '__pg_properties__', {}).keys())
 
     # *: filter for those with matching dictionary properties
     for key in set(args.keys()).intersection(pg_props):
         val = args[key]
-        val = val if isinstance(val, list) else [val]
-        if val:
+
+        # val is always a list, but the list elements are treated differently based on
```

Comment:
`val` for [`DataNode`](https://github.com/uc-cdis/peregrine/blob/c36d7ac86ef483f530055edff92992da7d17ca6c/peregrine/resources/submission/graphql/node.py#L975) is not a list but a string, then `for...in` below unpacks the string into separate characters by mistake, causing issue [PXD-2482](https://ctds-planx.atlassian.net/browse/PXP-2482). If we have no intention to change the GraphQL schema, I propose to add this harmless line back:

```python
val = val if isinstance(val, list) else [val]
```

Any thoughts pls? @vpsx @rudyardrichter